app/addons/databases/components.js

```diff
@@ -78,14 +79,74 @@ class DatabaseTable extends React.Component {
     showPartitionedColumn: PropTypes.bool.isRequired
   };
 
-  createRows = (dbList) => {
-    return dbList.map((item, k) => {
-      return (
-        <DatabaseRow item={item} key={k} showPartitionedColumn={this.props.showPartitionedColumn}/>
-      );
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      /* List of group names that should be unfolded */
+      openGroups: []
+    };
+  }
+
+  createRows = (dbList, nestLevel) => {
+    return dbList.map((item) => {
+      return (Array.isArray(item))
+        ? this.createGroup(item)
+        : (
+          <DatabaseRow
+            nestLevel={nestLevel || 0}
+            item={item}
+            key={item.id}
+            showPartitionedColumn={this.props.showPartitionedColumn}
+          />
+        );
     });
   };
 
+  createGroup = (items, nestLevel) => {
+    /* cannot create an empty group */
+    if (items.length === 0) {
+      /* @TODO add some error logging */
+      return;
+    }
+
+    const { openGroups } = this.state;
+
+    const groupName = this.getDatabaseGroupName(items[0]);
+    const isOpen = openGroups.includes(groupName);
+
+    const arrowIconClass = isOpen ? 'fonticon-up-dir' : 'fonticon-down-dir';
+
+    return [
+      <tr
+        key={`${groupName}-group-title`}
+        className="database-group-title"
+        onClick={() => this.toggleGroupVisibility(groupName)}
+      >
+        <td colSpan="4">
```

Comment:
I copied this from the database load error row, but I think it could be an issue with many`getExtensionColumns()`.